### PR TITLE
CARRY: implement e2e happy path

### PR DESCRIPTION
**Pull request by tedhtchang**
**What this PR does / why we need it**:
Add e2e SFTTrainer happy path test
**Which issue(s) this PR fixes** _(optional, in `Fixes #<issue number>, #<issue number>, ...` format, will close the issue(s) when PR gets merged)_:
Fixes https://issues.redhat.com/browse/RHOAIENG-4791

Test:
clone this PR
```sh
test/e2e/kind.sh
test/e2e/setup.sh
go test ./test/e2e
```
verify output:
```
ok  	github.com/kubeflow/training-operator/test/e2e	62.724s
```

### CODE REVIEW

File: test/e2e/kfto_kueue_sft_test.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+								},
+							},
+						},
+					},
+				}
+			},
+		},
+	}
+	// Need to implement the test client for the pytorchJob
+	// tuningjob, err := test.Client().kvtov1().pytorchJob().create()
```
Comment (tedhtchang) [inline test/e2e/kfto_kueue_sft_test.go:123]: @z103cb @astefanutti I am trying to wrap my head around how this is implemented..

Comment (astefanutti) [reply test/e2e/kfto_kueue_sft_test.go:123]: @tedhtchang this should be added into the test client. I've created https://github.com/project-codeflare/codeflare-common/pull/38. Once it's merged you should be able to bump the codeflare-common version and uncomment this.

File: test/e2e/kfto-kueue-sft.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
+apiVersion: v1
```
Comment (astefanutti) [inline test/e2e/kfto-kueue-sft.yaml:1]: Is this file really necessary now?

File: test/e2e/kfto_kueue_sft_test.go (hunk 1 above)
Comment (tedhtchang) [reply test/e2e/kfto_kueue_sft_test.go]: @astefanutti Thanks! I was able to create a pytorchjob.

**Review by astefanutti — APPROVED** [review]
LGTM! Awesome!

**Comment by astefanutti** [comment]
@tedhtchang Could you just add the `CARRY:` prefix to you commit please?
